elastic/kibana
# [Deployment management] Mark devtools and management sidenav deeplinks as visible on serverless

Fixes: https://github.com/elastic/kibana/issues/160349

## Summary

Recently there has been some work to use deeplink ids for the sidenav instead of plain hrefs to avoid broken links [#159125](https://github.com/elastic/kibana/pull/159125#discussion_r1227935144). This works goes over our `devtools` and `management` apps and conditionally sets the deeplinks as visible for the navbar if on serverless, otherwise we keep the default value (which is `hidden`).

This work follows a similar approach as taken by the observability team: https://github.com/elastic/kibana/pull/160620.

##### How to test
* Boot up any of the serverless solutions and make sure that both the `devtools` and `management` sections from the navbar are `hidden: false` and that they have `navLinkStatus: 1`.
* Boot up classic kibana and make sure that the left navbar has the same menu entries it always had.

To double check that the sections and links are not hidden and can be visible on navbar, edit the following file `packages/shared-ux/chrome/navigation/src/ui/components/navigation.tsx` and do the following:

```typescript
export function Navigation({ children, unstyled = false, dataTestSubj }: Props) {
  // const { onProjectNavigationChange, activeNodes$ } = useNavigationServices();
  const { onProjectNavigationChange, activeNodes$, navLinks$ } = useNavigationServices();
  const deepLinks = useObservable(navLinks$, []);
  console.log(deepLinks);
```

Then check in the browser devtools for the printed array of deeplinks, find `devtools` or `management` and make sure the sections aren't hidden.

<details>
<summary>Screenshots</summary>

![Screenshot 2023-07-05 at 06 53 00](https://github.com/elastic/kibana/assets/1191206/e0598a7d-7fbb-40d7-b790-0b549d25d058)
![Screenshot 2023-07-05 at 06 52 52](https://github.com/elastic/kibana/assets/1191206/6218b034-a42f-4e12-b811-3bbde5248da4)

</details>
